.github/workflows/docs.yml

```
@@ -0,0 +1,52 @@
+name: Docs
+
+on:
+  # Update on pushes to main
+  push:
+    branches: ['main']
+
+  # Allow running workflow manually
+  workflow_dispatch:
+
+# Restrict permissions to just Github Pages
+permissions:
+  contents: read
+  pages: write
+  id-token: write
+
+# Only run one at a time, but don't cancel in-progress deployments
+concurrency:
+  group: 'pages'
+  cancel-in-progress: false
+
+jobs:
+  # Build the site and publish an artifact
+  build-and-deploy:
+    runs-on: ubuntu-latest
+    environment:
+      name: github-pages
+      url: ${{ steps.deployment.outputs.page_url }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Setup JDK 17
+        uses: actions/setup-java@v4
+        with:
+          java-version: 17
+          distribution: 'temurin'
+          cache: 'gradle'
+
+      - name: Build documentation
+        run: |
+          chmod +x ./gradlew
+          ./gradlew dokkaHtml
```

```suggestion
        run: ./gradlew dokkaHtml
```

You no longer need to change the permissions of `gradlew` to make it executable, it now is by default. This can be condensed :)